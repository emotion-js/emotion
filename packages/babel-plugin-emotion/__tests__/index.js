import babelTester from 'babel-tester'
import plugin from 'babel-plugin-emotion'

babelTester('@emotion/babel-plugin-core', __dirname, {
  // we add a duplicate of our own plugin
  // users may run babel twice, and our plugin should be smart enough to not duplicate fields
  plugins: [plugin, [plugin, undefined, 'emotion-copy']]
})
