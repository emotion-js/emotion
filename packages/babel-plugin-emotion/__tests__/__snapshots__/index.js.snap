// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@emotion/babel-plugin-core core-with-css-import 1`] = `
"/** @jsx jsx */
import { jsx, css } from '@emotion/core'

const SomeComponent = props => (
  <div
    css={css\`
      color: hotpink;
    \`}
    {...props}
  />
)


      ↓ ↓ ↓ ↓ ↓ ↓

/** @jsx jsx */
import { jsx, css } from '@emotion/core';

var _ref = process.env.NODE_ENV === \\"production\\" ? {
  name: \\"1wl9j1m-SomeComponent\\",
  styles: \\"color:hotpink;label:SomeComponent;\\"
} : {
  name: \\"1wl9j1m-SomeComponent\\",
  styles: \\"color:hotpink;label:SomeComponent;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImNvcmUtd2l0aC1jc3MtaW1wb3J0LmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUtZIiwiZmlsZSI6ImNvcmUtd2l0aC1jc3MtaW1wb3J0LmpzIiwic291cmNlc0NvbnRlbnQiOlsiLyoqIEBqc3gganN4ICovXG5pbXBvcnQgeyBqc3gsIGNzcyB9IGZyb20gJ0BlbW90aW9uL2NvcmUnXG5cbmNvbnN0IFNvbWVDb21wb25lbnQgPSBwcm9wcyA9PiAoXG4gIDxkaXZcbiAgICBjc3M9e2Nzc2BcbiAgICAgIGNvbG9yOiBob3RwaW5rO1xuICAgIGB9XG4gICAgey4uLnByb3BzfVxuICAvPlxuKVxuIl19 */\\"
};

const SomeComponent = props => <div css={_ref} {...props} />;"
`;

exports[`@emotion/babel-plugin-core does-not-change-other-props 1`] = `
"/** @jsx jsx */
import { jsx } from '@emotion/core'

const Svg = <rect style={{ stroke: '#ff0000' }} />


      ↓ ↓ ↓ ↓ ↓ ↓

/** @jsx jsx */
import { jsx } from '@emotion/core';
const Svg = <rect style={{
  stroke: '#ff0000'
}} />;"
`;

exports[`@emotion/babel-plugin-core import-namespace-does-not-throw 1`] = `
"import * as emotionCore from '@emotion/core'
// could probably support this for real but it's pretty rare so I'm not gonna spend time on it now


      ↓ ↓ ↓ ↓ ↓ ↓

import * as emotionCore from '@emotion/core'; // could probably support this for real but it's pretty rare so I'm not gonna spend time on it now"
`;

exports[`@emotion/babel-plugin-core jsx-hoist 1`] = `
"/** @jsx jsx */
import { jsx } from '@emotion/core'

const SomeComponent = props => (
  <div
    css={{
      color: 'hotpink'
    }}
    {...props}
  />
)


      ↓ ↓ ↓ ↓ ↓ ↓

/** @jsx jsx */
import { jsx } from '@emotion/core';

var _ref = process.env.NODE_ENV === \\"production\\" ? {
  name: \\"1wl9j1m-SomeComponent\\",
  styles: \\"color:hotpink;label:SomeComponent;\\"
} : {
  name: \\"1wl9j1m-SomeComponent\\",
  styles: \\"color:hotpink;label:SomeComponent;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImpzeC1ob2lzdC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFLSSIsImZpbGUiOiJqc3gtaG9pc3QuanMiLCJzb3VyY2VzQ29udGVudCI6WyIvKiogQGpzeCBqc3ggKi9cbmltcG9ydCB7IGpzeCB9IGZyb20gJ0BlbW90aW9uL2NvcmUnXG5cbmNvbnN0IFNvbWVDb21wb25lbnQgPSBwcm9wcyA9PiAoXG4gIDxkaXZcbiAgICBjc3M9e3tcbiAgICAgIGNvbG9yOiAnaG90cGluaydcbiAgICB9fVxuICAgIHsuLi5wcm9wc31cbiAgLz5cbilcbiJdfQ== */\\"
};

const SomeComponent = props => <div css={_ref} {...props} />;"
`;

exports[`@emotion/babel-plugin-core static-object-with-camel-case 1`] = `
"/** @jsx jsx */
import { jsx } from '@emotion/core'

const SomeComponent = props => (
  <div
    css={{
      color: 'hotpink',
      backgroundColor: 'green'
    }}
    {...props}
  />
)


      ↓ ↓ ↓ ↓ ↓ ↓

/** @jsx jsx */
import { jsx } from '@emotion/core';

var _ref = process.env.NODE_ENV === \\"production\\" ? {
  name: \\"lcrht-SomeComponent\\",
  styles: \\"color:hotpink;background-color:green;label:SomeComponent;\\"
} : {
  name: \\"lcrht-SomeComponent\\",
  styles: \\"color:hotpink;background-color:green;label:SomeComponent;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0YXRpYy1vYmplY3Qtd2l0aC1jYW1lbC1jYXNlLmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUtJIiwiZmlsZSI6InN0YXRpYy1vYmplY3Qtd2l0aC1jYW1lbC1jYXNlLmpzIiwic291cmNlc0NvbnRlbnQiOlsiLyoqIEBqc3gganN4ICovXG5pbXBvcnQgeyBqc3ggfSBmcm9tICdAZW1vdGlvbi9jb3JlJ1xuXG5jb25zdCBTb21lQ29tcG9uZW50ID0gcHJvcHMgPT4gKFxuICA8ZGl2XG4gICAgY3NzPXt7XG4gICAgICBjb2xvcjogJ2hvdHBpbmsnLFxuICAgICAgYmFja2dyb3VuZENvbG9yOiAnZ3JlZW4nXG4gICAgfX1cbiAgICB7Li4ucHJvcHN9XG4gIC8+XG4pXG4iXX0= */\\"
};

const SomeComponent = props => <div css={_ref} {...props} />;"
`;

exports[`@emotion/babel-plugin-core static-object-with-child-selectors 1`] = `
"/** @jsx jsx */

import { jsx } from '@emotion/core'

const SomeComponent = props => {
  return (
    <div
      css={{
        color: 'green',
        ':hover': {
          color: 'hotpink'
        }
      }}
      {...props}
    />
  )
}


      ↓ ↓ ↓ ↓ ↓ ↓

/** @jsx jsx */
import { jsx } from '@emotion/core';

var _ref = process.env.NODE_ENV === \\"production\\" ? {
  name: \\"1y9ypgi-SomeComponent\\",
  styles: \\"color:green;:hover{color:hotpink;}label:SomeComponent;\\"
} : {
  name: \\"1y9ypgi-SomeComponent\\",
  styles: \\"color:green;:hover{color:hotpink;}label:SomeComponent;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0YXRpYy1vYmplY3Qtd2l0aC1jaGlsZC1zZWxlY3RvcnMuanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBT00iLCJmaWxlIjoic3RhdGljLW9iamVjdC13aXRoLWNoaWxkLXNlbGVjdG9ycy5qcyIsInNvdXJjZXNDb250ZW50IjpbIi8qKiBAanN4IGpzeCAqL1xuXG5pbXBvcnQgeyBqc3ggfSBmcm9tICdAZW1vdGlvbi9jb3JlJ1xuXG5jb25zdCBTb21lQ29tcG9uZW50ID0gcHJvcHMgPT4ge1xuICByZXR1cm4gKFxuICAgIDxkaXZcbiAgICAgIGNzcz17e1xuICAgICAgICBjb2xvcjogJ2dyZWVuJyxcbiAgICAgICAgJzpob3Zlcic6IHtcbiAgICAgICAgICBjb2xvcjogJ2hvdHBpbmsnXG4gICAgICAgIH1cbiAgICAgIH19XG4gICAgICB7Li4ucHJvcHN9XG4gICAgLz5cbiAgKVxufVxuIl19 */\\"
};

const SomeComponent = props => {
  return <div css={_ref} {...props} />;
};"
`;
